### ISSUE TITLE
Remove Dead Code: also removes variable assignments used in compact function

### ISSUE THREAD

**Issue by proclame** [laravel-shift/workbench-issues#21]
**Desktop:**
 - OS: Mac
 - OS Version: Big Sur
 - App Version: 0.4.1

**Describe the bug**
When running the new "Remove Dead Code" worker, a variable assignment that is used in a compact() call afterwards is also removed.

**Before**
```php
$transports = $driver->transports();

return view('drivers.processTransports', compact(['driver', 'transports', 'driverLinks', 'period']));
```

**After**
```php
$driver->transports();

return view('drivers.processTransports', compact(['driver', 'transports', 'driverLinks', 'period']));
```

**Comment by jasonmccreary** [comment]
@jessarcher, I believe you solved this before. Maybe a regression?

**Comment by jessarcher** [comment]
It turns out Rector only supports PHP's `compact(string $var_name, string ...$var_names)` syntax and ignores the alternative array syntax that is being used here.

I have PR'd Rector to support the full recursive array syntax which will hopefully be merged and released soon.

**Comment by jasonmccreary** [comment]
@proclame this should be resolved in the latest release (0.4.6). Let us know.

**Comment by proclame** [comment]
@jasonmccreary Confirmed! I ran it again (with the new version) on the same codebase, and worked as expected now!
Thanks @jessarcher :-)
